config/opal_config_asm.m4

```diff
@@ -864,21 +864,21 @@ AC_DEFUN([OPAL_CONFIG_ASM],[
 
     AC_ARG_ENABLE([builtin-atomics],
       [AC_HELP_STRING([--enable-builtin-atomics],
-         [Enable use of __sync builtin atomics (default: disabled)])])
-
+         [Enable use of __sync builtin atomics (default: enabled)])],
+         [], [enable_builtin_atomics="yes"])
     AC_ARG_ENABLE([osx-builtin-atomics],
       [AC_HELP_STRING([--enable-osx-builtin-atomics],
-         [Enable use of OSX builtin atomics (default: disabled)])])
+         [Enable use of OSX builtin atomics (default: enabled)])],
+         [], [enable_osx_builtin_atomics="yes"])
 
-    if test "$enable_builtin_atomics" = "yes" ; then
-       OPAL_CHECK_SYNC_BUILTINS([opal_cv_asm_builtin="BUILTIN_SYNC"],
-         [AC_MSG_ERROR([__sync builtin atomics requested but not found.])])
-       AC_DEFINE([OPAL_C_GCC_INLINE_ASSEMBLY], [1],
-         [Whether C compiler supports GCC style inline assembly])
+    opal_cv_asm_builtin="BUILTIN_NO"
+    if test "$opal_cv_asm_builtin" = "BUILTIN_NO" -a "$enable_builtin_atomics" = "yes" ; then
```

Comment:
Isn't the first part of this `test` trivially true because of the line above?  Is this just to match the style of the other `if` checks below so that the `if` block could be more easily moved around later?